src/algebra/monoid_algebra.lean

```diff
@@ -540,6 +545,43 @@ lemma single_mul_single {a₁ a₂ : G} {b₁ b₂ : k} :
 
 section
 
+/-- Like `finsupp.map_domain_add`, but for the convolutive multiplication we define in this file -/
+lemma map_domain_mul {α : Type*} {β : Type*} {α₂ : Type*} [semiring β] [add_monoid α] [add_monoid α₂] {x y : add_monoid_algebra β α}
+  (f : α → α₂) (hf : ∀ x y, f x + f y = f (x + y)):
+  (map_domain f (x * y : add_monoid_algebra β α) : add_monoid_algebra β α₂) =
+    (map_domain f x * map_domain f y : add_monoid_algebra β α₂) :=
+begin
+  simp only [mul_def, map_domain_sum, map_domain_single],
+  conv_rhs {
+    for (finsupp.sum _ _) [2] {
+      rw finsupp.sum_map_domain_index _ _,
+      skip,
+      exact λ a, by { simp, },
+      exact λ a b c, by { simp [mul_add], }
+    },
+    for (finsupp.sum _ _) [1] {
+      rw finsupp.sum_map_domain_index _ _,
+      skip,
+      exact λ a, by { simp, },
+      exact λ a b c, by { simp [add_mul], },
+    },
+  },
```

Comment:
I'd appreciate help simplifying this - I can't get `finsupp.sum_map_domain_index` to match without going into conv mode, and then I end up needing `exact` in conv mode to close the goals.